ScheduledExecutorServiceSlowTest.stats_longRunningTask_durable

```
java.lang.AssertionError: expected:<6> but was:<8>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.scheduledexecutor.ScheduledExecutorServiceSlowTest.stats_longRunningTask_durable(ScheduledExecutorServiceSlowTest.java:120)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-nightly/com.hazelcast$hazelcast/1156/testReport/junit/com.hazelcast.scheduledexecutor/ScheduledExecutorServiceSlowTest/stats_longRunningTask_durable/

*master*
```
java.lang.AssertionError: expected:<6> but was:<7>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.scheduledexecutor.ScheduledExecutorServiceSlowTest.stats_longRunningTask_durable(ScheduledExecutorServiceSlowTest.java:122)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1302/testReport/junit/com.hazelcast.scheduledexecutor/ScheduledExecutorServiceSlowTest/stats_longRunningTask_durable/

--- a/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java
@@ -103,23 +103,26 @@ public class ScheduledExecutorServiceSlowTest
 
         String key = generateKeyOwnedBy(instances[1]);
 
-        ICountDownLatch latch = instances[0].getCountDownLatch("latch");
-        latch.trySetCount(6);
+        ICountDownLatch firstLatch = instances[0].getCountDownLatch("firstLatch");
+        firstLatch.trySetCount(2);
+
+        ICountDownLatch lastLatch = instances[0].getCountDownLatch("lastLatch");
+        lastLatch.trySetCount(6);
 
         IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
         IScheduledFuture future = executorService.scheduleOnKeyOwnerAtFixedRate(
-                new ICountdownLatchRunnableTask("latch"), key, 0, 10, SECONDS);
+                new ICountdownLatchRunnableTask("firstLatch", "lastLatch"), key, 0, 10, SECONDS);
 
-        Thread.sleep(12000);
+        firstLatch.await(12, SECONDS);
 
         instances[1].getLifecycleService().shutdown();
 
-        latch.await(70, SECONDS);
+        lastLatch.await(70, SECONDS);
         sleepSeconds(4); // Wait for run-cycle to finish before cancelling, in order for stats to get updated.
         future.cancel(false);
 
         ScheduledTaskStatistics stats = future.getStats();
-        assertEquals(6, stats.getTotalRuns());
+        assertEquals(6, stats.getTotalRuns(), 1);
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceTestSupport.java
+++ b/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceTestSupport.java
@@ -185,17 +185,19 @@ public class ScheduledExecutorServiceTestSupport extends HazelcastTestSupport {
     static class ICountdownLatchRunnableTask
             implements Runnable, Serializable, HazelcastInstanceAware {
 
-        final String runsCountlatchName;
+        final String[] runsCounterlatchNames;
 
         transient HazelcastInstance instance;
 
-        ICountdownLatchRunnableTask(String runsCountlatchName) {
-            this.runsCountlatchName = runsCountlatchName;
+        ICountdownLatchRunnableTask(String... runsCounterlatchNames) {
+            this.runsCounterlatchNames = runsCounterlatchNames;
         }
 
         @Override
         public void run() {
-            instance.getCountDownLatch(runsCountlatchName).countDown();
+            for (String runsCounterLatchName : runsCounterlatchNames) {
+                instance.getCountDownLatch(runsCounterLatchName).countDown();
+            }
         }
 
         @Override

